q2_types/feature_data/_format.py

```diff
@@ -185,10 +185,19 @@ def sniff(self):
     AlignedDNAFASTAFormat)
 
 
+class DifferentialFormat(model.TextFileFormat):
```

Comment:
Are these _features_? If so, it might be helpful to leverage the `qiime2.Metadata` object for this (see [q2-demux ](https://github.com/qiime2/q2-demux/blob/master/q2_demux/_format.py#L64-L83)for an example).

Reply:
Yes!  That's a great idea -- I didn't realize that was in q2 yet.

Reply:
Although, the metadata columns will be different depending on the run.
For instance, if the user is simultaneously testing for differences between `age` and `sex`, they would have two columns corresponding to `age` and `sex` -- so the column names aren't necessarily known ahead of time.

Reply:
Variable/unknown columns are fine. I guess in that case, the thing to verify is
a) does the file [load/validate](https://github.com/qiime2/q2-demux/blob/master/q2_demux/_format.py#L74-L77) as metadata
b) is there at least [one column present](https://dev.qiime2.org/latest/api-reference/metadata/#qiime2.metadata.Metadata.column_count)